File: src/main/java/com/zhantu/entity/Product.java
package com.zhantu.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("product")
public class Product {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String code;
    private String name;
    private String subName;
    private String oem;
    private BigDecimal price;
    private Integer stock;
    private Long categoryId;
    private String series;
    private String qualityGrade;
    private String viscosity;
    private String spec;
    private String unit;
    private String brand;
    private String image;
    private String description;
    private String vinCompatible;
    private String oeNumber;
    private String groupCode;
    private Integer status;
    private Integer sales;
    @TableField(exist = false)
    private Integer favoriteCount;
    @TableLogic
    private Integer deleted;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: android-clients/app/src/main/java/com/zhantu/autopartsmall/data/model/Product.kt
package com.zhantu.autopartsmall.data.model

import com.google.gson.annotations.SerializedName

data class Product(
    val id: Long,
    val code: String,
    val name: String,
    val subName: String?,
    val oem: String?,
    val price: Double,
    val stock: Int,
    val categoryId: Long?,
    val series: String?,
    val qualityGrade: String?,
    val viscosity: String?,
    val spec: String?,
    val unit: String?,
    val brand: String?,
    val image: String?,
    val status: Int,
    val sales: Int?,
    @SerializedName("createTime") val createTimeRaw: Any?
) {
    val createTime: String?
        get() = when (createTimeRaw) {
            is String -> createTimeRaw
            is List<*> -> createTimeRaw.joinToString("-") { it.toString() }
            else -> null
        }

    fun getImageUrl(): String? {
        if (image.isNullOrEmpty()) return null
        val firstImage = image.split(",").firstOrNull() ?: return null
        return if (firstImage.startsWith("http")) firstImage else "http://10.178.93.31:8081$firstImage"
    }
}
